src/instanceApi.js

```diff
@@ -236,6 +236,23 @@ function instanceApi (oboeBus, contentSource) {
     source: contentSource
   }
 
+  function unsupportedOperation () {
+    throw new Error('Native Promises are not available and no Promise Library, via oboe.Promise, has been provided. Promise API cannot be supported.')
+  }
+
+  if (oboe.Promise || (typeof Promise !== 'undefined' && Promise.toString().indexOf('[native code]') !== -1)) {
```

Comment:
A bug source I thought about: what if I override the global `Promise` object? I often see Bluebird recommending to be used in your code as `Promise`, so I wouldn't be surprised if someone did some version of `window.Promise = require('bluebird')`. I think then the Promise body wouldn't include `[native code]`, so Oboe would then throw unsupported errors, unless the user also passed Bluebird in as `oboe.Promise`.

I think we want to encourage users towards explicitly passing in a promise library, but if this is a source of possible issues in the future, it would be worth handling it now.

Can you think of a good way to handle that?

Reply:
We could break out the checks `typeof Promise !== 'undefined'` and `Promise.toString().indexOf('[native code]') !== -1`. If the first passes and the second fails we can throw an error telling them to use oboe.Promise.

On the other hand we could just remove `Promise.toString().indexOf('[native code]') !== -1` and use what ever is provided.

We can add it to the docs that either way it needs to be A+ compatible.